File: revisiting_ctci/1_9_StringRotation.py
import unittest



def IsRotation(a,b):
	""" take strings a and b as an input and see if they are rotations
		of one another, returns boolean """
	rotation = 0
	rotate_max = len(a)

	while rotation < rotate_max:
		rotation += 1

		if a == b:
			return True

		a = a[-1] + a[:-1]

	return False


class TestRotate(unittest.TestCase):
	def test_IsRotation(self):
		self.assertTrue(IsRotation('test','ttes'))
		self.assertTrue(IsRotation('test','estt'))

		self.assertFalse(IsRotation('test','sett'))
		self.assertFalse(IsRotation('test','nottest'))

if __name__ == '__main__':


	unittest.main()

File: revisiting_ctci/1_9_StringRotation.go
package main

import (
	"fmt"
)

func IsRotation(a string, b string) bool {

	for i := 0; i < len(a); i++ {

		if a == b {
			return true
		}

		a = fmt.Sprintf("%v%v", string(a[len(a)-1]), a[:(len(a)-1)])

	}

	return false
}

func main() {

	fmt.Println("test and ttes are rotations:")
	fmt.Println(IsRotation("test", "ttes"))

	fmt.Println("test and sett are rotations:")
	fmt.Println(IsRotation("test", "sett"))

	fmt.Println("test and estt are rotations:")
	fmt.Println(IsRotation("test", "estt"))

	fmt.Println("test and golang are rotations:")
	fmt.Println(IsRotation("test", "golang"))

}
